src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/nametags/NametagEnchantmentRenderer.kt

```diff
@@ -60,37 +57,62 @@ private object EnchantmentDisplayHelper {
         )
     }
     
+    private fun getEnchantmentName(enchantment: RegistryKey<Enchantment>): String {
+        val idPath = enchantment.value.toString().substringAfter(':')
+        val translationKey = "enchantment.minecraft.$idPath"
+        return I18n.translate(translationKey)
+    }
+    
+    private fun getSingleWordAbbreviation(word: String): String = word.take(3)
+    
+    private fun getInitialsAbbreviation(words: List<String>): String = 
+        words.joinToString("") { it.first().toString() }
+    
+    private fun getCompoundAbbreviation(words: List<String>): String {
+        val firstWord = words[0]
+        
+        if (firstWord.length >= 3) {
+            return firstWord.take(3)
+        }
+        
+        val remainingChars = 3 - firstWord.length
+        return firstWord + words.getOrNull(1)?.take(remainingChars).orEmpty()
+    }
+    
+    private fun processMultiWordName(words: List<String>): String {
+        if (words.size == 1) {
+            return getSingleWordAbbreviation(words[0])
+        }
+        
+        val initials = getInitialsAbbreviation(words)
+        if (initials.length >= 3) {
+            return initials
+        }
+        
+        return getCompoundAbbreviation(words)
+    }
```

Comment:
This can be optimized further (see below):
```suggestion
    private fun processMultiWordName(words: List<String>): String {
        val initials = getInitialsAbbreviation(words)
        
        return if (initials.length >= 3) {
            initials
        } else {
            getCompoundAbbreviation(words)
        }
    }
```